docs/engineering_design.rst

```diff
@@ -3,8 +3,149 @@
+
+Nevertheless, this section aims to provide a summary of the models
+design. As of writing this docs, there are five distinct new models in this
+app. However, when looking at the logical design we would find only three
+models like the following:
+
+The ``CourseAccessGroup`` Model
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is the main model which specify which learners should have
+access to which courses.
+
+This model has two Many to Many relationships:
+
+ * **Course:** Courses that belongs to this group. To simplify the use
+   relation with the Django REST Framewor it has been moved to its
+   own model named ``GroupCourse``.
+
+ * **User:** Users who are member of this group. For the same reasons above
+   it has it's own model named ``Membership``.
+
+This model is the only essential model for this app to run. The others can
+be thought as complementary.
+
+The ``PublicCourse`` Model
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+This model is used to mark courses as public to exempt from the
+Course Access Group rules.
+
+Instead of modifying ``CourseOverview`` to have an additional boolean field
+e.g. ``CourseOverview.cag_is_public``, a new model has been created.
+
+This necessary to avoid having complex database migration design and
+minimize the maintenance impact of this app.
+
+The ``MembershipRule`` Model
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+This model is used to automatically assign learners to Course Access Groups.
+It utilizes the
+`USER_ACCOUNT_ACTIVATED <https://github.com/edx/edx-platform/pull/23296>`_
+Django signal to match learners into groups based on their email domain name.
+
+
+Future Plans
+------------
+
+This application works on a *modified* fork of the Hawthorn release. This
+section denotes plans to support future releases the impact on the
+architecture of this app.
+
+Supporting Bridgekeeper in Juniper
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As of April, 2019 the Open edX team started to use
+`Bridgekeeper <https://github.com/edx/edx-platform/pull/20250>`_ for Access
+Control which eventually would deprecate the ``courseware.has_access``
+function.
+
+The plan to support Bridgekeeper is to remove the hooks for ``has_access``
+and replace them with ``bridgekeeper.perms[]`` rules.
+
+For more information about Bridgekeeper check the project documentation:
+https://bridgekeeper.readthedocs.io/.
+
+This is probably the change with the most impact. So far there's no concrete
+plan to adapt to it.
+
+In addition to this documentation please consider taking a look at the
+`Access Control Backends thread`_ on Open edX Discuss.
+It touches on a couple of related topics regarding Bridgekeeper and some
+recommendation from the edX engineers.
+
+Single-site Setups
+~~~~~~~~~~~~~~~~~~
+The majority of the Open edX Platform installations are single-site setups
+in which Site Configurations isn't used. This application doesn't support
+such installations at the moment. In order to support this installation
```

Comment:
suggest a comma after "installation" or rephrase to "Several modifications need to be done to support this installation."